amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectorSettings.scala

```diff
@@ -153,22 +154,52 @@ object TemporaryQueueSourceSettings {
     TemporaryQueueSourceSettings(connectionProvider, exchange)
 }
 
+final class AmqpPublishConfirmConfiguration private (
+    val confirmTimeout: FiniteDuration
+) {
+  override def toString: String =
+    "AmqpPublishConfirmConfiguration(" +
+    s"confirmTimeout=$confirmTimeout" +
+    ")"
+}
+
+object AmqpPublishConfirmConfiguration {
+  def apply(confirmTimeout: FiniteDuration): AmqpPublishConfirmConfiguration =
+    new AmqpPublishConfirmConfiguration(confirmTimeout)
+
+  def unapply(settings: AmqpPublishConfirmConfiguration): Option[FiniteDuration] =
+    Some(settings.confirmTimeout)
+
+  /**
+   * Java API
+   */
+  def create(confirmTimeout: FiniteDuration): AmqpPublishConfirmConfiguration =
```

Comment:
For the Java API please use `java.time.Duration` and convert it with `akka.util.JavaDurationConverters._`.